frontend/vitest.config.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2025)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { coverageConfigDefaults, defineConfig } from "vitest/config"
+
+export default defineConfig({
+  test: {
+    // Define workspace paths directly
+    workspace: [
+      "app/vite.config.ts",
+      "lib/vite.config.ts",
+      "connection/vite.config.ts",
+      "utils/vite.config.ts",
+    ],
+
+    // Global coverage configuration
+    coverage: {
+      provider: "v8",
+      reporter: ["text-summary", "json-summary", "html"],
+      include: [
+        "app/src/**/*",
+        "lib/src/**/*",
+        "utils/src/**/*",
+        "connection/src/**/*",
+      ],
```

Comment:
There is starting to be an effort for any new package to include these files in multiple locations. Is this something that could be resolved with a single `packages` directory by chance?

Reply:
You mean that we need to include the path for all new sub-packages here? I think we could also just use slighlty more generic globs here, e.g.:
```
include: ["*/src/**/*"]
```

and above:

```
 workspace: [
      "*/vite.config.ts"
    ],
```

This would probably allow to get all packages picked up automatically. We would just need to manage more specific excludes here for what to not include in the coverage.

Reply:
I'm fine with it as is for now, I am not sure when we are adding new packages, so it might be fine. Was just curious if there's a way to make this simpler so that when I add a package I don't accidentally forget to add it somewhere.